### PR TITLE
Migrated README to Markdown (PEP 566)

### PR DESCRIPTION
**Pull request by 9999years**
Migrated `README.rst` to `README.md` and updated `setup.py` to match;
this is in line with [PEP 566](https://www.python.org/dev/peps/pep-0566/#description-content-type-optional) and the new `long_description_content_type`
arg in setuptools.

Also updated `MANIFEST.in` to include the new `README.md` with `include *.md` as suggested by `check-manifest` — is that the right way to do that?

See also:

 * [Relevant section in PEP 566](https://www.python.org/dev/peps/pep-0566/#description-content-type-optional)
 * [The CommonMark specification](http://spec.commonmark.org)
 * [The Description-Content-Type field documentation in the Python
   packaging documentation](https://packaging.python.org/specifications/core-metadata/#description-content-type-optional)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -57,6 +57,13 @@
     # https://packaging.python.org/specifications/core-metadata/#description-optional
     long_description=long_description,  # Optional
 
+    # Denotes that our long_description is in Markdown; valid values are
+    # text/plain, text/x-rst, and text/markdown
+    #
+    # This field corresponds to the "Description-Content-Type" metadata field:
+    # https://packaging.python.org/specifications/core-metadata/#description-content-type-optional
+    long_description_content_type='text/markdown',  # Optional
```
Comment (di) [inline setup.py]: Might want to add a note here that while this field is technically optional, it's required if you want a Markdown `long_description`.

**Comment by 9999years** [comment]
Added a comment to clarify:

> Optional if `long_description` is written in RST but required for plain-text or Markdown; if unspecified, "applications should attempt to render [the long_description] as `text/x-rst; charset=UTF-8` and fall back to `text/plain` if it is not valid rst" ([link](https://packaging.python.org/specifications/core-metadata/#description-optional))

File: setup.py [hunk 2]
```diff
@@ -53,10 +53,22 @@
     # Often, this is the same as your README, so you can just read it in from
     # that file directly (as we have already done above)
     #
+    # Optional if long_description is written in RST but required for
+    # plain-text or Markdown; if unspecified, "applications should attempt to
+    # render [the long_description] as text/x-rst; charset=UTF-8 and fall back
+    # to text/plain if it is not valid rst" (see link below)
+    #
```
Comment (di) [inline setup.py]: I think this is in the wrong place, it should be below `long_description_content_type=...`.

Comment (9999years) [reply setup.py]: Yeah, noticed just after pushing. Should be updated now.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# A sample Python project
+
+A sample project that exists as an aid to the [Python Packaging User
+Guide][packaging guide]’s [Tutorial on Packaging and Distributing
```
Comment (di) [inline README.md]: Could you replace the `’` character with `'` here?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+Projects][distribution tutorial].
+
+This projects does not aim to cover best practices for Python project
```
Comment (di) [inline README.md]: There's a typo here that's not your fault, should be "this project".

File: setup.py [hunk 5]
```diff
@@ -57,6 +57,18 @@
     # https://packaging.python.org/specifications/core-metadata/#description-optional
     long_description=long_description,  # Optional
 
+    # Denotes that our long_description is in Markdown; valid values are
+    # text/plain, text/x-rst, and text/markdown
+    #
+    # Optional if long_description is written in rst but required for
```
Comment (di) [inline setup.py]: Could you change `rst` here to `reStructuredText`? Since newcomers might not know what "rst" stands for.

File: setup.py [hunk 6]
```diff
@@ -57,6 +57,18 @@
...
+    # plain-text or Markdown; if unspecified, "applications should attempt to
+    # render [the long_description] as text/x-rst; charset=UTF-8 and fall back
+    # to text/plain if it is not valid rst" (see link below)
+    #
+    # This field corresponds to the "Description-Content-Type" metadata field:
+    # https://packaging.python.org/specifications/core-metadata/#description-content-type-optional
+    long_description_content_type='text/markdown',  # Optional
```
Comment (di) [inline setup.py]: Maybe change `# Optional` here to `# Optional (see note above)`?

**Comment by 9999years** [comment]
OK, fixed.

**Comment by 9999years** [comment]
Can we get this merged now that pypa/warehouse#869 is closed?
Thanks!

**Comment by di** [comment]
Thanks @9999years!

**Pull request merged**